### ISSUE TITLE
Stimulus Reflex very slow

### ISSUE THREAD

**Issue by colby-swandale** [rubyapi/rubyapi#1119]
I'm finding the Signature toggle to be very slow in production, ~5s to toggle the signatures on/off, compared to development where it is instant.

**Comment by colby-swandale** [comment]
This is also maybe what's causing browsers to not have an established WebSocket connection when the user toggles the Type Signatures. 

See https://github.com/rubyapi/rubyapi/pull/1110#issuecomment-1154673870

**Comment by natematykiewicz** [comment]
<img width="801" alt="Screen Shot 2022-06-14 at 8 52 59 AM" src="https://user-images.githubusercontent.com/5104186/173594489-a6863ac5-6dec-4c5b-9ead-04b4c664b2b8.png">
My problem is the Content Security Policy.

```
Content-Security-Policy: default-src 'self' https:; font-src 'self' https: data:; img-src 'self' https: data:; object-src 'none'; script-src 'self' https:; style-src 'self' https: 'unsafe-inline'; connect-src 'self' https:
```

Notice the `; connect-src 'self' https:`. I think the problem is that `wss:` protocol is not permitted by the content security policy. Though, I've never used ActionCable.

**Comment by natematykiewicz** [comment]
For example, https://github.com/rubyapi/rubyapi/blob/main/config/initializers/content_security_policy.rb.

In development you've got:
```ruby
policy.connect_src :self, :https, "http://localhost:3035", "ws://localhost:3035"
```

But in production you've got:
```ruby
policy.connect_src :self, :https
```

I think production might need to change.

**Comment by natematykiewicz** [comment]
Or does the line need to go away entirely? My config/initializers/content_security_policy.rb at work is commented out entirely. But the only `connect_src` that's in it is this:

```ruby
# If you are using webpack-dev-server then specify webpack-dev-server host
policy.connect_src :self, :https, "http://localhost:3035", "ws://localhost:3035" if Rails.env.development?
```

Sounds like 3035 is for webpack-dev-server's hot reload

**Comment by colby-swandale** [comment]
> Or does the line need to go away entirely?

I don't believe so. If we remove the line, I expect Chrome and Firefox will start to refuse to connect to the WebSocket. Safari doesn't particularly like `self` or `https` for some odd reason.

**Comment by natematykiewicz** [comment]
That fixed my CSP errors for Safari and Firefox. Now I think the problem is just that we're sending over 1MB of HTML through a websocket every time you press the toggle.

I'm not sure what all you're toggling, but it might be wise to just always include the type signatures in the HTML and just show/hide them via JS, then cache that HTML in Cloudflare globally.
